lib/sqlite-manager.js

```diff
@@ -9,1031 +9,1017 @@
+    // Get the ndarray keys
+    const ndarrayKeys = findCollectionKeys(schema, sqliteConstants.SQLITE_GENERAL_TYPE_NDARRAY);
+    if (ndarrayKeys.length > 0)
+      // eslint-disable-next-line require-atomic-updates
+      result.data = await sqliteNdarray.readNdarrayMany(db, result.data, ndarrayKeys);
   }
 
-  /**
-   * Updates data in a dataset resource.
-   * @alias module:sqlite-manager.updateData
-   * @async
-   * @param {object} db - The sqlite3 db object from module node-sqlite3.
-   * @param {DataRow|DataRow[]} data - The data to update.
-   *     Must conform to the schema defined by the resource metadata.
-   *     Supports updating individual or multiple rows.
-   * @param {boolean} [upsert=false] - Indicates the data should be created if no
-   *     document/row is found matching the primary key.
-   * @param {boolean} [throws=true] - Indicates whether this function should reject
-   *     if there is an error. The TDX-API doesn't, as it returns a field which
-   *     states if there has been an error.
-   * @returns {Promise<CommandResult>} - Use the result property to check for
-   *     errors.
-   */
-  manager.updateData = async function(db, data, upsert, throws) {
-    // See https://www.sqlite.org/lang_UPSERT.html for more info.
-
-    // setup default values
-    let set_upsert = upsert;
-    if (set_upsert === undefined) {
-      set_upsert = false;
-      throw Error("upsert === false is not supported yet.");
-    }
-    let set_throws = throws;
-    if (set_throws === undefined) {
-      set_throws = true;
-    }
+  return result;
+}
 
-    const schema = await manager.getGeneralSchema(db);
-    const dataToConvert = [].concat(data);
-    const sqlData = dataToConvert.map((row) => {
-      return sqliteConverter.convertRowToSqlite(schema, row);
-    });
-    const infoTable = await sqliteInfoTable.getInfoKeys(db, ["schema"]);
-    const uniqueIndex = infoTable[0].schema.uniqueIndex;
+/**
+ * Updates data in a dataset resource.
+ * @alias module:sqlite-manager.updateData
+ * @async
+ * @param {object} db - The sqlite3 db object from module node-sqlite3.
+ * @param {DataRow|DataRow[]} data - The data to update.
+ *     Must conform to the schema defined by the resource metadata.
+ *     Supports updating individual or multiple rows.
+ * @param {boolean} [upsert=false] - Indicates the data should be created if no
+ *     document/row is found matching the primary key.
+ * @param {boolean} [throws=true] - Indicates whether this function should reject
+ *     if there is an error. The TDX-API doesn't, as it returns a field which
+ *     states if there has been an error.
+ * @returns {Promise<CommandResult>} - Use the result property to check for
+ *     errors.
+ */
+module.exports.updateData = async function(db, data, upsert, throws) {
+  // See https://www.sqlite.org/lang_UPSERT.html for more info.
+
+  // setup default values
+  let set_upsert = upsert;
+  if (set_upsert === undefined) {
+    set_upsert = false;
```

Comment:
Why set to false if you're throwing an error?

Reply:
It was code I was going to add, then didn't get the time to.

I've deleted it, since it's just confusing until @mereacre or I do implement it.